v2/cmd/asoctl/internal/importing/resource_import_result.go

```diff
@@ -22,7 +24,71 @@ type ResourceImportResult struct {
 	resources []genruntime.MetaObject
 }
 
+// Count returns the number of successfully imported resources.
+func (r *ResourceImportResult) Count() int {
+	return len(r.resources)
+}
+
 func (r *ResourceImportResult) SaveToWriter(destination io.Writer) error {
+	return r.writeTo(r.resources, destination)
+}
+
+func (r *ResourceImportResult) SaveAllToFile(filepath string) error {
+	return r.saveTo(r.resources, filepath)
+}
+
+func (r *ResourceImportResult) SaveToFolder(folder string) error {
+	// We name the files after the resource type and name
+	// We allocate resources to files using a map, just in case we have a naming collision
+	// (If that happens, all the similarly named resources will be in the same file, which is not ideal,
+	// but better than dropping one or more)
+	fileMap := make(map[string][]genruntime.MetaObject, len(r.resources))
+	for _, resource := range r.resources {
+		resourceName := resource.GetName()
+		typeName := resource.GetObjectKind().GroupVersionKind().Kind
```

Comment:
You could include the group in here too, in a `group-kind-name`, which would guarantee uniqueness and avoid the awkward possibility of collisions entirely?

Reply:
Presumably if you did this, the map could just be `map[string]genruntime.MetaObject` I guess?

Reply:
I think we'd still need a `map[string][]genruntime.MetaObject` because we can (in theory) follow ARM URL references into different subscriptions. I'd rather gather all the resources (and force the user to resolve the conflicts in the yaml files) than non-deterministically drop one of them.